ee/api/role.py

```diff
@@ -0,0 +1,110 @@
+from typing import cast
+
+from django.db import IntegrityError
+from rest_framework import mixins, serializers, viewsets
+from rest_framework.permissions import SAFE_METHODS, BasePermission, IsAuthenticated
+
+from ee.models.role import Role, RoleMembership
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.api.shared import UserBasicSerializer
+from posthog.models import OrganizationMembership
+from posthog.models.user import User
+from posthog.permissions import OrganizationMemberPermissions, extract_organization
+
+DEFAULT_ROLE_NAME = "Write"
+
+
+class RolePermissions(BasePermission):
+    """
+    Requires organization admin level to change object, allows everyone read
+    """
+
+    message = "You need to have admin level or higher."
+
+    def has_permission(self, request, view):
+        organization = extract_organization(request.user)
```

Comment:
oof I don't like `extract_organization` here!

Since we know `request.user` is a `User`, can we just do `request.user.organization` ?

`extract_organization` seems to have been created for cases where we don't know the model that its being used on.